### PR TITLE
Feat - NX conversion

### LINKED ISSUE #1777: Switch to eslint as tslint is deprecated
### Do you want to request a *feature* or report a *bug*?

Bug

#### Bug Report

StackBlitz template: https://stackblitz.com/edit/covalent

#### Which version of Angular and Material, and which browser and OS does this issue affect?

Did this work in previous versions of Angular / Material?

### PR DESCRIPTION
**Pull request by owilliams320**
## Description
In an effort to modernize covalent and allow this repository to continue to grow. We have re-organized this library and supporting documentation app to support a mono repo style development flow using NX dev tools. Most of the changes in this pr address a few critical key points to support sustainability for the future.

### What's included?
- Addition of NX build tools and nx.cloud see [nx.dev](https://nx.dev/) | [nx.cloud](https://nx.app/)
- New folder structure that clearly separates library modules from application type modules
- Support of using ESLint over TsLint. - closes #1777
- Support of using Jest as a test runner to dramatically increase test speed
- Use of TS Strict mode! this will help avoid typos and other type checking in angular templates [angular.io/strict-mode](https://angular.io/guide/strict-mode)

### Where did it all go

| Module | Before | Now |
|:---:|:---:|:---:|
| Covalent Docs | `src/apps` | `apps/docs` |
| @covalent/core | `src/platform/core` | `libs/angular` |
| @covalent/code-editor | `src/platform/code-editor` | `libs/angular-code-editor` |
| @covalent/dynamic-forms | `src/platform/dynamic-forms` | `libs/angular-dynamic-forms` |
| @covalent/echarts | `src/platform/echarts` | `libs/angular-echarts` |
| @covalent/guided-tour | `src/platform/guided-tour` | `libs/angular-guided-tour` |
| @covalent/highlight | `src/platform/highlight` | `libs/angular-highlight` |
| @covalent/text-editor | `src/platform/text-editor` | `libs/angular-text-editor` |
| @covalent/markdown | `src/platform/markdown` | `libs/markdown` |
| @covalent/flavored-markdown | `src/platform/flavored-markdown` | `libs/markdown-flavored` |
| @covalent/markdown-navigator | `src/platform/markdown-navigator` | `libs/markdown-navigator` |

#### Test Steps
- [x] `npm run build`
- [x] `npm run start`
- [x] then open localhost:4200 
- [x] finally run throught the entire Docs app to verify functionality is working as before

#### General Tests for Every PR

- [x] `npm run start` still works.
- [x] `npm run lint` passes.
- [x] `npm run stylelint` passes.
- [x] `npm test` passes and code coverage is not lower.
- [x] `npm run build` still works.

#### Sections to Review
- [x] Angular - Hai
- [x]  Code Editor - Josh
- [x]  Dynamic Forms - Anders
- [x]  Echarts - Josh
- [x]  Guided Tour - Mike
- [x]  Highlight - Hai
- [x]  Markdown - Hai
- [x]  Markdown flavored - Hai
- [x]  Markdown navigator - Hai

##### NX Todo
- [x]  Release step to distribute all packages to NPM (Github Action)
- [x]  Update release step to deploy GH Pages doc app
- [x]  Add a simple E2E test for docs app

### CODE REVIEW

**Comment by Josh-Schoen** [comment]
The editor in the docs app looks to be missing some styles. I tested this both on 3.x, 4.x locally with the same result so not sure what's exactly going on but clear it's not pulling styles in from `node_modules`.

Screen Shot 2022-02-25 at 1 23 24 PM

File: apps/docs-app/src/app/content/components/component-demos/guided-tour/demos/guided-tour-demo-go-back-navigation/guided-tour-demo-go-back-navigation.component.ts [hunk 1]
```diff
@@ -1,5 +1,9 @@
 import { Component, OnInit } from '@angular/core';
-import { CovalentGuidedTourService, IGuidedTour, ITourEvent } from '@covalent/guided-tour';
+import {
+  CovalentGuidedTourService,
+  IGuidedTour,
+  ITourEvent,
+} from '@covalent/guided-tour';
```
Comment (mikespudwebb) [inline apps/docs-app/src/app/content/components/component-demos/guided-tour/demos/guided-tour-demo-go-back-navigation/guided-tour-demo-go-back-navigation.component.ts:6]: step 4 of this demo never seems to get highlighted. Not the end of the world really :)

**Review by mikespudwebb — APPROVED** [review]
A Herculean effort!! Thanks for converting everything to use `nx`.

**Comment by mrjihai** [comment]
The search box has a weird border 
image


image

**Comment by mrjihai** [comment]
Flavored Markdown Loader
 "https://github.com/angular/angular/blob/main/README.md" is 404.
V4 uses "'https://github.com/angular/angular/blob/master/README.md".  
Not a bug. Just need to update the link in doc.
image
image

**Comment by DersJ** [comment]
Dynamic forms LGTM

**Comment by mrjihai** [comment]
V4
Screen Shot 2022-03-16 at 1 22 42 PM

Menu icon is missing. Will look into it later
Screen Shot 2022-03-16 at 1 22 55 PM

**Comment by mrjihai** [comment]
image
Cannot close sidesheet

**Comment by mrjihai** [comment]
image

Dynamic forms Custom Elements section is blank in both nx and V4.

V3
image

**Comment by owilliams320** [comment]
All issues mentioned are fixed! Thanks again everyone who contributed to this super large code change @mikespudwebb @mrjihai @DersJ @Josh-Schoen

**Pull request merged**